src/output-json.c

```diff
@@ -159,10 +164,6 @@ void EveFileInfo(JsonBuilder *jb, const File *ff, const uint64_t tx_id, const ui
             break;
     }
 
-    if (ff->flags & FILE_SHA256) {
-        jb_set_hex(jb, "sha256", (uint8_t *)ff->sha256, (uint32_t)sizeof(ff->sha256));
-    }
```

Comment:
This was always logged as even truncated, these files are written into the file store with this sha256 as its filename. So lets you associate this record with the logged file. Probably should have been a comment here.

Reply:
ok, I should have thought about that. I'm going to remove it.

Reply:
I'm wondering if if could help to do something like:
```diff
diff --git a/src/output-json.c b/src/output-json.c
index 4815665f1b..f3b74bc58c 100644
--- a/src/output-json.c
+++ b/src/output-json.c
@@ -149,6 +149,9 @@ void EveFileInfo(JsonBuilder *jb, const File *ff, const uint64_t tx_id, const ui
             if (ff->flags & FILE_SHA1) {
                 jb_set_hex(jb, "sha1", (uint8_t *)ff->sha1, (uint32_t)sizeof(ff->sha1));
             }
+            if (ff->flags & FILE_SHA256) {
+                jb_set_hex(jb, "sha256", (uint8_t *)ff->sha256, (uint32_t)sizeof(ff->sha256));
+            }
             break;
         case FILE_STATE_TRUNCATED:
             JB_SET_STRING(jb, "state", "TRUNCATED");
@@ -161,13 +164,12 @@ void EveFileInfo(JsonBuilder *jb, const File *ff, const uint64_t tx_id, const ui
             break;
     }
 
-    if (ff->flags & FILE_SHA256) {
-        jb_set_hex(jb, "sha256", (uint8_t *)ff->sha256, (uint32_t)sizeof(ff->sha256));
-    }
-
     if (flags & FILE_STORED) {
         JB_SET_TRUE(jb, "stored");
         jb_set_uint(jb, "file_id", ff->file_store_id);
+        if ((ff->state != FILE_STATE_CLOSED) && (ff->flags & FILE_SHA256)) {
+            jb_set_hex(jb, "sha256", (uint8_t *)ff->sha256, (uint32_t)sizeof(ff->sha256));
+        }
     } else {
         JB_SET_FALSE(jb, "stored");
         if (flags & FILE_STORE) {
```
where we log only if file is stored or complete ?

Reply:
My proposal is stupid: incomplete files are interesting and need a sha256 to find them. I'm going to close this PR soon and do an updated one without the patch.